march_rqt_robot_monitor/src/march_rqt_robot_monitor/diagnostic_analyzers/gait_state.py

```diff
@@ -0,0 +1,41 @@
+
+
+from diagnostic_msgs.msg import DiagnosticStatus
+import rospy
+
+from march_shared_resources.msg import GaitActionGoal
+
+
+class CheckGaitStatus(object):
+    def __init__(self, updater):
+        """Initializes an gait diagnostic which analyzes gait and subgait states.
+
+        :type updater: diagnostic_updater.Updater
+        """
+        self._updater = updater
+
+        self._goal_sub = rospy.Subscriber('/march/gait/schedule/goal', GaitActionGoal, self._cb_goal)
+
+    def _cb_goal(self, msg):
+        """Callback for the gait scheduler goal.
+
+        :param msg: GaitGoal
+        """
+        diagnostic = self._diagnostics(msg.goal.name, msg.goal.current_subgait)
+        self._updater.add('(Sub)gait status', diagnostic)
```

Comment:
Why didn't you create one diagnostic that is constantly updated? Just like in the temperature diagnostic.

Reply:
I was more or less using the same structure as the imc states because this would only require the creating of the class and not a whole subset of adding diagnostics. I found this works kind off nice and after this PR would translate all the other diagnostics (control and temperature) to the same structure. This would allow for a way nicer updater.py file which is better readable and more intuitive.

Reply:
I am getting this error everytime I want to look at the summary:
```
Traceback (most recent call last):
  File "/opt/ros/melodic/lib/python2.7/dist-packages/rqt_robot_monitor/inspector_window.py", line 125, in _signal_queue_updated
    status = self.timeline.get_all_status_by_name(self._name)
  File "/opt/ros/melodic/lib/python2.7/dist-packages/rqt_robot_monitor/timeline.py", line 173, in get_all_status_by_name
    return [status[name] for status in list(self._queue)]
KeyError: '/Gait/(Sub)gait status'
```

Reply:
took me some time to figure out what was going on but eventually fixed the problem.